sdk/core/ts-http-runtime/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -136,85 +31,25 @@ function getChallenge(response: PipelineResponse): string | undefined {
 export function bearerTokenAuthenticationPolicy(
   options: BearerTokenAuthenticationPolicyOptions,
 ): PipelinePolicy {
-  const { credential, scopes, challengeCallbacks } = options;
-  const logger = options.logger || coreLogger;
-  const callbacks = {
-    authorizeRequest: challengeCallbacks?.authorizeRequest ?? defaultAuthorizeRequest,
-    authorizeRequestOnChallenge: challengeCallbacks?.authorizeRequestOnChallenge,
-    // keep all other properties
-    ...challengeCallbacks,
-  };
+  const { credential, scopes } = options;
 
-  // This function encapsulates the entire process of reliably retrieving the token
-  // The options are left out of the public API until there's demand to configure this.
-  // Remember to extend `BearerTokenAuthenticationPolicyOptions` with `TokenCyclerOptions`
-  // in order to pass through the `options` object.
-  const getAccessToken = credential
-    ? createTokenCycler(credential /* , options */)
-    : () => Promise.resolve(null);
+  const getAccessToken = credential ? credential.getToken : () => Promise.resolve(null);
 
   return {
     name: bearerTokenAuthenticationPolicyName,
-    /**
-     * If there's no challenge parameter:
-     * - It will try to retrieve the token using the cache, or the credential's getToken.
-     * - Then it will try the next policy with or without the retrieved token.
-     *
-     * It uses the challenge parameters to:
-     * - Skip a first attempt to get the token from the credential if there's no cached token,
-     *   since it expects the token to be retrievable only after the challenge.
-     * - Prepare the outgoing request if the `prepareRequest` method has been provided.
-     * - Send an initial request to receive the challenge if it fails.
-     * - Process a challenge if the response contains it.
-     * - Retrieve a token with the challenge information, then re-send the request.
-     */
     async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
       if (!request.url.toLowerCase().startsWith("https://")) {
```

Comment:
I think we'll need to relax this. For 3p the scenario where devs are testing e2e in the local environment is going to be a lot more common.

Historically this has been controversial, however I think this can be a dealbreaker for 3p. Some of the options that have been discussed:

* Allow non https when allowInsecureConnection option is true
* Allow non https when the host is a wellknown local host such as localhost, 127.0.0.1, ::1, host.docker.internal 
* Above plus an environment variable - Caveat here is that env variables in browser are tricky

TBH I don't have a strong preference as long as we have a way to enable this.